src/SprykerEco/Zed/CrefoPayApi/Business/Logger/CrefoPayApiLogger.php

```diff
@@ -0,0 +1,144 @@
+{
+    use TransactionTrait;
+
+    protected const GET_REQUEST_METHOD = 'get%sRequest';
+    protected const GET_RESPONSE_METHOD = 'get%sResponse';
+
+    /**
+     * @var \SprykerEco\Zed\CrefoPayApi\Persistence\CrefoPayApiEntityManagerInterface
+     */
+    protected $entityManager;
+
+    /**
+     * @var string
+     */
+    protected $requestType;
+
+    /**
+     * @param \SprykerEco\Zed\CrefoPayApi\Persistence\CrefoPayApiEntityManagerInterface $entityManager
+     */
+    public function __construct(CrefoPayApiEntityManagerInterface $entityManager)
+    {
+        $this->entityManager = $entityManager;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     * @param \Generated\Shared\Transfer\CrefoPayApiResponseTransfer $responseTransfer
+     * @param string $requestType
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayApiLogTransfer
+     */
+    public function logApiCall(
+        CrefoPayApiRequestTransfer $requestTransfer,
+        CrefoPayApiResponseTransfer $responseTransfer,
+        string $requestType
+    ): PaymentCrefoPayApiLogTransfer {
+        $this->requestType = $requestType;
+        $paymentCrefoPayApiLog = $this->createPaymentCrefoPayApiLogTransfer(
+            $requestTransfer,
+            $responseTransfer
+        );
+
+        return $this->getTransactionHandler()->handleTransaction(function () use ($paymentCrefoPayApiLog) {
+            return $this->entityManager->savePaymentCrefoPayApiLog($paymentCrefoPayApiLog);
+        });
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     * @param \Generated\Shared\Transfer\CrefoPayApiResponseTransfer $responseTransfer
+     *
+     * @return \Generated\Shared\Transfer\PaymentCrefoPayApiLogTransfer
+     */
+    protected function createPaymentCrefoPayApiLogTransfer(
+        CrefoPayApiRequestTransfer $requestTransfer,
+        CrefoPayApiResponseTransfer $responseTransfer
+    ): PaymentCrefoPayApiLogTransfer {
+        return (new PaymentCrefoPayApiLogTransfer())
+            ->setRequestType($this->requestType)
+            ->setCrefoPayOrderId($this->getCrefoPayOrderId($requestTransfer))
+            ->setIsSuccess($responseTransfer->getIsSuccess())
+            ->setResultCode($this->getResultCode($responseTransfer))
+            ->setMessage($this->getMessage($responseTransfer))
+            ->setSalt($this->getSalt($responseTransfer))
+            ->setRequest($requestTransfer->serialize())
+            ->setResponse($responseTransfer->serialize());
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CrefoPayApiRequestTransfer $requestTransfer
+     *
+     * @return string
+     */
+    protected function getCrefoPayOrderId(CrefoPayApiRequestTransfer $requestTransfer): string
+    {
+        $method = sprintf(static::GET_REQUEST_METHOD, ucfirst($this->requestType));
+
+        return $requestTransfer->$method()->getOrderID();
```

Comment:
it would be great to check if we have this method in this transfer and throw an exception if not.